src/log.c

```diff
@@ -180,40 +125,23 @@ static int readHeader(Log *log)
     return RR_OK;
 }
 
-static int writeLength(void *buf, size_t cap, char prefix, long len)
-{
-    return safesnprintf(buf, cap, "%c%ld\r\n", prefix, len);
-}
-
-static int writeLong(void *buf, size_t cap, long val)
-{
-    int len = lensnprintf("%ld", val);
-    return safesnprintf(buf, cap, "$%d\r\n%ld\r\n", len, val);
-}
-
-static int writeString(void *buf, size_t cap, const char *val)
-{
-    int len = lensnprintf("%s", val);
-    return safesnprintf(buf, cap, "$%d\r\n%s\r\n", len, val);
-}
-
 static int writeEntry(Log *log, raft_entry_t *ety)
 {
     int rc;
-    size_t off = 0;
+    size_t n = 0;
     char buf[1024];
 
     size_t offset = FileSize(&log->file);
     size_t idxoffset = FileSize(&log->idxfile);
 
-    off += writeLength(buf + off, sizeof(buf) - off, '*', ENTRY_ELEM_COUNT);
-    off += writeString(buf + off, sizeof(buf) - off, ENTRY_STR);
-    off += writeLong(buf + off, sizeof(buf) - off, ety->term);
-    off += writeLong(buf + off, sizeof(buf) - off, ety->id);
-    off += writeLong(buf + off, sizeof(buf) - off, ety->type);
-    off += writeLength(buf + off, sizeof(buf) - off, '$', ety->data_len);
+    n += multibulkWriteLen(buf + n, sizeof(buf) - n, '*', ENTRY_ELEM_COUNT);
+    n += multibulkWriteStr(buf + n, sizeof(buf) - n, ENTRY_STR);
+    n += multibulkWriteLong(buf + n, sizeof(buf) - n, ety->term);
+    n += multibulkWriteLong(buf + n, sizeof(buf) - n, ety->id);
+    n += multibulkWriteLong(buf + n, sizeof(buf) - n, ety->type);
+    n += multibulkWriteLen(buf + n, sizeof(buf) - n, '$', (int) ety->data_len);
 
-    if (FileWrite(&log->file, buf, off) != (ssize_t) off ||
+    if (FileWrite(&log->file, buf, n) != (ssize_t) n ||
```

Comment:
again, dont understand the 0 length here.

Reply:
never mind, n, not 0.

Reply:
I just want to respect 80 column limit when it is possible, that was the reason for that adjustment. I added another commit, line is shorter, let me know if this is better.